### ISSUE TITLE
`x is not Ellipsis` does not narrow x

### ISSUE THREAD

**Issue by anabelle2001** [microsoft/pyright#10721]
Today, pyright does not narrow on `x is not Ellipsis`. It correctly narrows for `x is not ...`, and `not isinstance(x,types.EllipsisType)`. 

Code sample in [pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwKYGcoEsILCGUCiANoegiuigCqJIBQoksi6AdgOYZY54CGKKSXAH0WSAG6DatACZJgUYAAoAHgC4CxUuSo0AlKtpQjGeSzB4KrFDB4sAxkhUAaDSTIVqyfYeO%2B%2BA4VEJEBVdIwBiKAB5AGkAOilZeTYVdSI3bU8kb190eWUMNDM8ONKDXz9%2BQRgRcUFQiOj4xLkoAAtU1y0PPXLjPKgCiihirvcUPoqof2ra4IaoSJAkbFwAQRA2AFcIJBYYLPUAInTunWQjwpHzaf50NhYeACNCJFgwJmQoI4A5OpAjkA), [mypy playground](https://mypy-play.net/?mypy=latest&python=3.12&gist=d546bd0f30eda9a83a5e94debe3683a1)

```python
from types import EllipsisType
from typing import assert_never

def f(x: EllipsisType):
    if not isinstance(x, EllipsisType):
        assert_never(x)  # OK.

def g(x: EllipsisType):
    if x is not ...:
        assert_never(x)  # OK.

def h(x: EllipsisType):
    if x is not Ellipsis:
        assert_never(x)  # reportArgumentType: "EllipsisType" is not assignable to type "Never"
```

[^1]: https://docs.python.org/3/library/constants.html#Ellipsis

**Comment by erictraut** [comment]
The expression form `x is not Ellipsis` is not currently supported as a narrowing pattern whereas `isinstance(x, T)` and `x is ...` are. For a full list of supported type narrowing patterns, refer to [this documentation](https://microsoft.github.io/pyright/#/type-concepts-advanced?id=type-guards).

Each supported type narrowing form requires specialized logic, and there are countless possible forms so it's not possible to support them all. For that reason, type checkers tend to support only common idioms. Pyright already supports more type narrowing forms than any other Python type checker. We occasionally add new support for new forms, but this is done only if we have good evidence that it's a commonly-used pattern.

I'm going to change this from a bug report to a enhancement request. Then I'm going to close the issue because I don't have any intention of adding this feature at this time. If it receives sufficient upvotes, I will reconsider it in the future.
